### PR TITLE
nym-cli: support name-service contract

### LINKED ISSUE #3355: Support functionality for nym-name-service (nym-api, nym-cli, etc)

### PR DESCRIPTION
**Pull request by octol**
# Description

Closes: #3355

Add support to nym-cli for querying, registering, and deleting entries in the name service contract.

### CODE REVIEW

File: common/commands/src/validator/mixnet/query/query_all_names.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    #[clap(value_parser)]
+    #[clap(help = "Optionally, the registered name to display")]
+    pub name: Option<String>,
+}
+
+pub async fn query(args: Args, client: &QueryClientWithNyxd) {
+    match client.nym_api.get_registered_names().await {
+        Ok(res) => {
+            if let Some(name) = args.name {
+                let name = res.iter().find(|name_entry| {
```
Comment (jstuczyn) [inline common/commands/src/validator/mixnet/query/query_all_names.rs]: I wonder if perhaps the contract should let you query by name natively given it (I think) has indexes on those

**Review by jstuczyn — APPROVED** [review]

File: common/commands/src/validator/mixnet/query/query_all_names.rs (hunk 1 above)
Comment (octol) [reply common/commands/src/validator/mixnet/query/query_all_names.rs]: yep the contract allows you to do that, but this is the nym-api and currently it's only returning a list of all (cached) names

File: common/commands/src/validator/mixnet/query/query_all_names.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                        name_entry.name.address.to_string(),
+                        name_entry.name.name.to_string(),
+                    ]);
+                }
+
+                println!("The registered names in the directory are:");
+                println!("{table}");
+            }
+        }
+        Err(NymAPIError::NotFound) => {
```
Comment (jstuczyn) [inline common/commands/src/validator/mixnet/query/query_all_names.rs:48]: should this error be even possible?

**Review by jstuczyn — APPROVED** [review]

File: common/commands/src/validator/mixnet/query/query_all_names.rs (hunk 2 above)
Comment (octol) [reply common/commands/src/validator/mixnet/query/query_all_names.rs:48]: Currently on mainnet it does, for example

Comment (octol) [reply common/commands/src/validator/mixnet/query/query_all_names.rs:48]: Since that endpoint isn't (yet) deployed

**Pull request merged**